### ISSUE TITLE
Invalid response code 190 while trying to add the lock to Homey

### ISSUE THREAD

**Issue by LarsMachiels** [AdyRock/com.switchbot#73]
I have a Homey Pro (2023) and installed the Switchbot app (first the regular one and later the test version). I have the lock and the mini hub. Cloud service for the lock is activated. I keep on getting "Invalid response code: 190" when the app tries to add the lock. Login was ok. What could be the problem? Thanks in advance for any help.

P.s.: the curtain module E1 is working fine with the same app.

**Comment by pmha1** [comment]
Did you find a solution Lars ? I just bought the Smartlock and got the same annoying problem.

**Comment by pmha1** [comment]
Found it……this worked for me…

One thing you could try is to enable developer mode in the SwitchBot phone app.

a) Go to Profile > Preference
b) Tap App Version 10 times. Developer Options will show up

**Comment by IanAlbinson** [comment]
I'm getting this same issue when trying to add the Switchbot Temp and Humidity Sensor in Homey. I have set up the Developer options and can see the token and secret key.

Is there a solution available?

**Comment by AdyRock** [comment]
When you add a device it should ask you the first time to login to your SwitchBot account, where you can enter your SwitchBot username and password to authorise that app. Do you see that?

**Comment by IanAlbinson** [comment]
Yup, did all that, and then it went back to the app where it displayed the "Invalid response code 190" screen. Tried a few times, but the same things happens, unfortunately.

**Comment by AdyRock** [comment]
Have you enable cloud access for the device in the Switchbot app?

**Comment by IanAlbinson** [comment]
Where would I find that in the app? I don’t see it.

**Comment by AdyRock** [comment]
When you tap on the Device you can tap on the gear in the top right corner. There you will see the option 
![Screenshot_20240118_191707_SwitchBot.jpg](https://github.com/AdyRock/com.switchbot/assets/38184396/c32585a8-a40c-4966-b46c-9937dbbb3c39)

Make sure Cloud services is On.

**Comment by IanAlbinson** [comment]
Ah, thank you. Yes, Cloud services are on for all my devices, which are mainly the temp sensors and strip lighting.

**Comment by AdyRock** [comment]
When you logged in while adding the device, did you see any errors?

**Comment by IanAlbinson** [comment]
No.

I add it, it takes me to the login screen for Switchbot. I sign in, it says “You can now close this window.”, I do, and gives me the Invalid response code 190 error. See attached screenshot.

**Comment by IanAlbinson** [comment]
![IMG_4707](https://github.com/AdyRock/com.switchbot/assets/5090542/dfc02f2a-3200-4a2c-ba68-c2c74f14dccd)
